lib/strategy/FragmentationStrategyShape.ts

```diff
@@ -0,0 +1,452 @@
+   */
+  private readonly contentOfStorage: Set<string>;
+  /**
+ * @description The iris handled
+ */
+  private readonly irisHandled: Set<string> = new Set();
+
+  /**
+   * A random generator
+   */
+  private randomGenerator: prand.RandomGenerator;
+
+  /**
+   * A map of the shape name with a function to generate a shape Iri
+   */
+  public readonly shapeIriMap: Record<string, (podIri: string) => string>;
+
+  /**
+   * @description A map of the storage and the resource type handled
+   */
+  private readonly contentHandledByPod: Map<string, Set<string>> = new Map();
+
+  public static readonly SHAPE_INDEX_FILE_NAME: string = 'shapeIndex';
+
+  public static readonly SHAPE_INDEX_PREFIX = 'https://shapeIndex.com#';
+
+  public static readonly RDF_TYPE_NODE = DF.namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type');
+  public static readonly RDF_TRUE = DF.literal('true', DF.namedNode('http://www.w3.org/2001/XMLSchema#boolean'));
+  public static readonly RDF_STRING_TYPE = DF.namedNode('http://www.w3.org/2001/XMLSchema#string');
+
+  public static readonly SHAPE_INDEX_LOCATION_NODE = DF.namedNode(`${FragmentationStrategyShape.SHAPE_INDEX_PREFIX}shapeIndexLocation`);
+  public static readonly SHAPE_INDEX_CLASS_NODE = DF.namedNode(`${FragmentationStrategyShape.SHAPE_INDEX_PREFIX}ShapeIndex`);
+  public static readonly SHAPE_INDEX_ENTRY_NODE = DF.namedNode(`${FragmentationStrategyShape.SHAPE_INDEX_PREFIX}entry`);
+  public static readonly SHAPE_INDEX_BIND_BY_SHAPE_NODE = DF.namedNode(`${FragmentationStrategyShape.SHAPE_INDEX_PREFIX}bindByShape`);
+  public static readonly SHAPE_INDEX_DOMAIN_NODE = DF.namedNode(`${FragmentationStrategyShape.SHAPE_INDEX_PREFIX}domain`);
+  public static readonly SHAPE_INDEX_IS_COMPLETE_NODE = DF.namedNode(`${FragmentationStrategyShape.SHAPE_INDEX_PREFIX}isComplete`);
+
+  public static readonly SOLID_INSTANCE_NODE = DF.namedNode('http://www.w3.org/ns/solid/terms#instance');
+  public static readonly SOLID_INSTANCE_CONTAINER_NODE =
+  DF.namedNode('http://www.w3.org/ns/solid/terms#instanceContainer');
+
+  /**
+   * @param {Record<string, IShapeEntry>} shapeConfig - A map of the shape path
+   * where the values follow the interface IShapeEntry @range {json}
+   * @param {string[]} contentOfStorage - The content of a storage
+   * @param {string} relativePath - the relatif path of the IRI
+   * @param {boolean|undefined} tripleShapeTreeLocator - indicate if a shape locator triple is generated
+   * @param {string|undefined} randomSeed - the random seed for the selection of the shape template
+   */
+  public constructor(
+    shapeConfig: Record<string, IShapeEntry>,
+    contentOfStorage: string[],
+    relativePath?: string,
+    tripleShapeTreeLocator?: boolean,
+    randomSeed?: number,
+  ) {
+    super();
+    if (contentOfStorage.length === 0) {
+      throw new Error('there should be at least one content type in the resource');
+    }
+    if (randomSeed === undefined) {
```

Comment:
Do you still need this random stuff in here now that you have the other fragmenter?

Reply:
Yes, because we can select a shape from a set of shapes if it is defined in the config.